### PR TITLE
ci: create clair PR when new claircore tag is created

### PR DESCRIPTION
**Pull request by crozzy**
Something we have to do over and over is update clair with the latest claircore tag. This automates some of that workflow.

TODO:

- [x] How to authenticate? (with an app)

### CODE REVIEW

File: .github/workflows/update-clair.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        with:
+          add-paths: go.mod,go.sum
+          token: ${{ steps.generate_token.outputs.token }}
+          base: main
+          commit-message: |
+            'chore: update claircore to ${{ env.VERSION }}
+            
+            Update claircore to latest tag
+            '
+          committer: No One <noreply@quay.io>
```
Comment (hdonnay) [inline .github/workflows/update-clair.yml:50]: Can this be something like `Claircore-CI <${{ github.actor }}@users.noreply.github.com>`?

File: .github/workflows/update-clair.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        id: cpr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          add-paths: go.mod,go.sum
+          token: ${{ steps.generate_token.outputs.token }}
+          base: main
+          commit-message: |
+            'chore: update claircore to ${{ env.VERSION }}
+            
+            Update claircore to latest tag
```
Comment (hdonnay) [inline .github/workflows/update-clair.yml:48]:
```suggestion
            Update claircore to latest release.
```

File: .github/workflows/update-clair.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+on:
+  push:
+    tags:
+      - v*.*.*
```
Comment (hdonnay) [inline .github/workflows/update-clair.yml:4]:
```suggestion
      - v1.*.*
```

We're going to rev this at some point.

**Review by hdonnay — CHANGES_REQUESTED** [review]
Some minor nits.

I think this is fine as-is. Do you have any strong feelings on patch-only updates vs minor and patch?

File: .github/workflows/update-clair.yml (hunk 3 above)
Comment (crozzy) [reply .github/workflows/update-clair.yml:4]: Not sure what the point in limiting to v1 tags is? Would we not want this behavior when it's iterated to v2 v3 etc (assuming we don't have a better mechanism by then)?

Comment (crozzy) [reply .github/workflows/update-clair.yml:4]: I suppose I just thought it'd create PRs for everything and we'd just close what we didn't want. What would be the advantage of paring down?

Comment (hdonnay) [reply .github/workflows/update-clair.yml:4]: Yeah, fair enough.

**Review by hdonnay — DISMISSED** [review]

**Review by hdonnay — APPROVED** [review]

**Pull request merged**